drools/src/test/java/com/datatorrent/drools/operator/StatelessDroolsOperatorTest.java

```diff
@@ -98,4 +118,119 @@ public void testSpringActivation()
 //    }
 
   }
+
+  protected List<Object> getListForRuleMatch()
+  {
+    List<Object> factList = new ArrayList<>();
+    factList.add(new Product(1,"gold",0));
+    factList.add(new Product(2,"gold",0));
+    factList.add(new Product(3,"gold",0));
+    factList.add(new Product(4,"gold",0));
+
+    factList.add(new Product(3,"diamond",0));
+    factList.add(new Product(3,"diamond",0));
+    factList.add(new Product(3,"diamond",0));
+
+    return factList;
+  }
+
+  protected List<Object> getListForNoRuleMatch()
+  {
+    List<Object> factList = new ArrayList<>();
+    factList.add(new Product(1,"fakeGold",0));
+    factList.add(new Product(21,"fakeGold",0));
+    factList.add(new Product(3,"fakeGold",0));
+    factList.add(new Product(4,"fakeGold",0));
+
+    factList.add(new Product(3,"fakeDiamond",0));
+    factList.add(new Product(3,"fakeDiamond",0));
+    factList.add(new Product(3,"fakeDiamond",0));
+
+    return factList;
+  }
+
+  @Test
+  public void testListenerRulesMatch()
+  {
+    testMeta.underTest.setLoadSpringSession(true);
+    testMeta.underTest.setSessionName("test-session");
+    testMeta.underTest.activate(testMeta.context);
+    Assert.assertEquals(0,factsAndFiredRulesPort.collectedTuples.size());
+    Assert.assertEquals(0,factsPort.collectedTuples.size());
+    Assert.assertEquals(0,ruleCountPort.collectedTuples.size());
+    List<Object> factsList = getListForRuleMatch();
+    testMeta.underTest.beginWindow(0);
+    for (Object fact : factsList) {
+      testMeta.underTest.factsInput.process(fact);
+    }
+    Assert.assertEquals(0,factsPort.collectedTuples.size());
+    testMeta.underTest.endWindow();
+    Assert.assertEquals(1,ruleCountPort.collectedTuples.size());
+    Assert.assertEquals(1,factsAndFiredRulesPort.collectedTuples.size());
+    //It is expected to get 8 tuples as platinum is added via rules
```

Comment:
Inserted fact could be different test. It's a good idea to write a test with one specific test purpose. It's okay to have multiple small tests. When something goes wrong it's easy to nail down if we have separate tests.